R/emtr.R

```diff
@@ -1,10 +1,50 @@
 parameters_from_file <- function(parameters_file) {
-  parameters_df <- openxlsx::read.xlsx(parameters_file)
-  parameters <- parameters_from_df(parameters_df)
+  params_type <- tools::file_ext(parameters_file)
+  if (params_type %in% c("xlsx", "xls")) {
+    parameters_df <- openxlsx::read.xlsx(parameters_file, sheet = 1) %>% as.data.frame()
+    parameters <- parameters_from_df(parameters_df)
+  } else if (params_type %in% c("yaml", "yml")) {
+    parameters <- yaml::read_yaml(parameters_file)
+  } else {
+    error_msg <- paste0("Unsupported params type: ", params_type)
+    stop(error_msg)
+  }
+  # Rename any missing app parameters if possible
+  renamed_app_params <- c(
+    "FamilyAssistance_FTC_Rates_FirstChild" = "FamilyAssistance_FTC_Rates_FirstChild0to15",
+    "FamilyAssistance_FTC_Rates_SubsequentChild" = "FamilyAssistance_FTC_Rates_SecondChild0to12",
+    "FamilyAssistance_IWTC_Eligibility" = "FamilyAssistance_IWTC_IncomeTest",
+    "FamilyAssistance_IWTC_IncomeThreshold_Couple" = "FamilyAssistance_IWTC_IncomeThreshold",
+    "FamilyAssistance_IWTC_IncomeThreshold_Single" = "FamilyAssistance_IWTC_IncomeThreshold"
+  )
+  for (app_param in names(renamed_app_params)) {
+    if (!(app_param %in% names(parameters))) {
+      tawa_param <- renamed_app_params[[app_param]]
+      if (tawa_param %in% names(parameters)) {
+        print(sprintf("Copying %s to %s", tawa_param, app_param))
+        parameters[[app_param]] <- parameters[[tawa_param]]
+      } else {
+        error_msg <- sprintf(
+          "Missing parameter, neither app (%s) nor tawa (%s) parameter found ",
+          app_param, tawa_param
+        )
+        stop(error_msg)
+      }
+    }
+  }
+  # Add defaults for any missing system params
+  default_system_params <- parameters_from_df(as.data.frame(PARAMS_TEMPLATE))

```

Comment:
Fill missing parameters with value 0?

Reply:
This fills missing parameters with their default value, which might not be 0, could be 1 or even a string.